src/promise-katas-1.js

```diff
@@ -30,86 +30,143 @@ Remember, a Promise can only resolve successfully, or reject when it fails to re
+    resolve(["eggs", "apples", "milk", "bread"]);
+  });
+};
 
 // 6 Create a function that returns a promise which returns "An error occurred" if it rejects
 // Example:
 // anError() rejects to "An error occurred"
 
-const anError = () => { };
+const anError = () => {
+  return new Promise((resolve, reject) => {
+    reject("An error occurred");
+  });
+};
 
 // 7 Create a function that returns a promise which returns 666 if it rejects
 // Example:
 // theNumberOfTheBeast() rejects to the value 666
 
-const theNumberOfTheBeast = () => { };
+const theNumberOfTheBeast = () => {
+  return new Promise((resolve, reject) => {
+    reject(666);
+  });
+};
 
 // 8 Create a function that returns a promise which returns an object with a key of error and value of 500 if it rejects
 // Example:
 // internalServerError() rejects to { error: 500 }
 
-const internalServerError = () => { };
+const internalServerError = () => {
+  return new Promise((resolve, reject) => {
+    reject({ error: 500 });
+  });
+};
 
-// 9 Create a function that returns a promise which resolves to "happy" if the parameter is greater than 
+// 9 Create a function that returns a promise which resolves to "happy" if the parameter is greater than
 // or equal to 1 and rejects to "sad" if the value is less than 1
 // Example:
 // happySad(1) resolves to "happy"
 // happySad(0) rejects to "sad"
 
-const happySad = () => { };
-
-// 10 Create a function that returns a promise which resolves to "Yes. Luke, I am your father." if the 
+const happySad = (x) => {
+  return new Promise((resolve, reject) => {
+    //const happySad(num);
+    if (x >= 1) {
+      resolve("happy");
+    } else {
+      reject("sad");
+    }
+  });
+};
+
+// 10 Create a function that returns a promise which resolves to "Yes. Luke, I am your father." if the
 // parameter is "Luke" and rejects to "Not your dad." if the value is anything else
 // Example:
 // amIYourFather() resolves to "Yes. Luke, I am your father."
 // amIYourFather() rejects to "Not your dad."
 
-const amIYourFather = () => { };
+const amIYourFather = (stringName) => {
+  return new Promise((resolve, reject) => {
+    stringName === "Luke"
+      ? resolve("Yes. Luke, I am your father.")
+      : reject("Not your dad.");
+  });
+};
 
-// 11 Create a function that returns a promise. The promise should resolve to a new function that returns "My name is <input name>" 
+// 11 Create a function that returns a promise. The promise should resolve to a new function that returns "My name is <input name>"
 // Example:
 // A call to myNameIs() resolves to a new function that takes one argument, name, and returns "My name is name" Eg (name) => `My name is ${name}`
 
-const myNameIs = () => { };
-
+const myNameIs = () => {
+  return new Promise((resolve, reject) => {
+    const inputName = () => {
+      const inputName2 = () => {
+        return `My name is Romy`;
+      };
+      return inputName2;
+    };
+    resolve(inputName());
+  });
+};
```

Comment:
Theres a bit of over complication here, you can remove the two nested function declarations and just return a single function in the resolve block. Also, you need it to accept a name in the first place and this currently doesnt have that so will always return the hard coded value. This would do the trick:
```
const myNameIs = () => {
  return new Promise((resolve, reject) =>
    resolve((name) => `My name is ${name}`)
  );
};
```